src/Renderer/QueryResultsRenderer.ts

```diff
@@ -545,6 +547,64 @@ export class QueryResultsRenderer {
         }
     }
 
+    private processTaskLinks(task: Task): Task {
+        // Skip if task is from the same file as the query
+        if (this.filePath === task.path) {
+            return task;
+        }
+
+        const linkCache = task.file.cachedMetadata.links;
+        if (!linkCache) return task;
+
+        // Find links in the task description
+        const taskLinks = linkCache.filter((link) => {
+            return (
+                link.position.start.line === task.taskLocation.lineNumber &&
+                task.description.includes(link.original) &&
+                link.link.startsWith('#')
+            );
+        });
+        if (taskLinks.length === 0) return task;
+
+        let description = task.description;
+
+        // a task can only be from one file, so we can replace all the internal links
+        //in the description with the new file path
+        for (const link of taskLinks) {
+            // Find the occurrence closest to the original position in the source document
+            let currentIndex = 0;
+            let bestMatch = -1;
+            let closestDistance = Number.MAX_SAFE_INTEGER;
+
+            let foundIndex = description.indexOf(link.original, currentIndex);
+            while (foundIndex !== -1) {
+                const distance = Math.abs(foundIndex - link.position.start.col);
+                if (distance < closestDistance) {
+                    bestMatch = foundIndex;
+                    closestDistance = distance;
+                }
+
+                currentIndex = foundIndex + 1; // Move past this occurrence
+                foundIndex = description.indexOf(link.original, currentIndex);
+            }
+
+            // Skip if no match was found (should not happen in practice)
+            if (bestMatch === -1) continue;
+
+            const fullLink = `[[${task.path}${link.link}|${link.displayText}]]`;
+            description =
+                description.slice(0, bestMatch) + fullLink + description.slice(bestMatch + link.original.length);
+        }
+
```

Comment:
I am really sorry that [my comment](https://github.com/obsidian-tasks-group/obsidian-tasks/pull/3357#issuecomment-2707227304) came after you had done this work.

I do not have the time to do all the manual testing required to verify all the many different scenarios in Tasks that would be affected by this - some of which I listed in that comment.

So yes, please do revert this back to the previous behaviour that I approved. Thanks in advance.

Reply:
No worries- it was an unfortunate timing in communications